### PR TITLE
hide charts for some filter options

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

File: mod/metrics/lib/card/metric.rb [hunk 1]
```diff
@@ -64,45 +64,58 @@ class << self
...
+      opts[:metric_type] ||= :researched
+      opts[:value_type] ||= "Number" if opts[:metric_type] == :researched
+      opts[:metric_type] = Card.fetch_name opts[:metric_type]
+
+      opts.each_with_object({}) do |(field, content), subfields|
+        type_id = subfield_type_id(field)
+        content = Array.wrap(content).to_pointer_content if type_id == Card::PointerID
+        subfields[field] = { content: content,
+                             type_id: type_id }
+      end
```
Comment (ethn) [inline mod/metrics/lib/card/metric.rb]: would be nice to break this into more methods.

File: mod/metrics/lib/card/metric.rb [hunk 2]
```diff
@@ -64,45 +64,58 @@ class << self
     # @option opts [Boolean] :random_source (false) pick a random source for
     #   each value
     def create opts, &block
-      opts[:type] ||= :researched
-      metric = Card.create! name: opts[:name],
+      random_source = opts.delete :random_source
```
Comment (ethn) [inline mod/metrics/lib/card/metric.rb:69]: not super comfortable with having "random_source" as the most prominent variable.

File: mod/metrics/set/abstract/04_chart.rb [hunk 3]
```diff
@@ -22,6 +22,33 @@ def value_to_highlight
...
+
+  def chart_filter_hash
+    card.filter_hash(zoom_in?)
+  end
+
+  def zoom_in?
+    card.numeric? # && chart_item_count > 10
+  end
+end
+
```
Comment (ethn) [inline mod/metrics/set/abstract/04_chart.rb:51]: I like these.

File: mod/metrics/set/type/metric/thumbnail.rb [hunk 4]
```diff
@@ -18,7 +18,11 @@ def thumbnail_title
                                    view: :name),
                         class: "ellipsis")
     return content unless voo.show?(:thumbnail_link)
-    link_to_card card, content, title: card.metric_title_card.name
+    opts = { title: card.metric_title_card.name }
+    if voo.closest_live_option(:project)
+      opts[:path] = { filter: { project: voo.closest_live_option(:project) } }
+    end
```
Comment (ethn) [inline mod/metrics/set/type/metric/thumbnail.rb:24]: oy, I can't quite remember what this API means.

File: mod/wikirate/set/self/project.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  view :core do
+    total = [0, 0, 0]
+    Card.search(type_id: card.id).map do |pc|
+      cnts = [pc.num_metrics, pc.num_companies, pc.num_answers, pc.num_users]
+      3.times do |i|
+        total[i] += cnts[i]
+      end
+      CSV.generate_line([pc.name] + cnts + [pc.num_policies])
```
Comment (ethn) [inline mod/wikirate/set/self/project.rb:14]: maybe a method for the row?

File: mod/wikirate/set/type/project.rb [hunk 6]
```diff
@@ -19,6 +19,23 @@ def num_metrics
   @num_metrics ||= metric_card.item_names.size
 end
 
+def num_users
+  Answer.where(where_answer).where("updated_at > ?", created_at).select(:creator_id).uniq.count
+end
+
+def num_answers
+  Answer.where(where_answer).where("updated_at > ?", created_at).count
+end
```
Comment (ethn) [inline mod/wikirate/set/type/project.rb]: Is there already an `answers` method?

**Review by ethn — APPROVED** [review]
Good stuff.

File: mod/metrics/set/type/metric/thumbnail.rb (hunk 4 above)
Comment (xithan) [reply mod/metrics/set/type/metric/thumbnail.rb:24]: With the introduction of your voo API it suddenly wasn't possible anymore to pass arguments to other views. `closest_live_option` was my emergency hack to get that feature back.

**Pull request merged**